File: src/fdStatistics_old.m
function fdStats = fdStatistics_old(fdData,discardMode)
% Copyright (c) 2017 Cai Jiashen

fdStats.evalSlices = numel(fdData);

[quot, rem] = quorem(sym(fdStats.evalSlices),2);

if discardMode == false
    first = 1;
    last = fdStats.evalSlices;
    fdStats.usedSlices = fdStats.evalSlices;
elseif discardMode == true
    first = 2;
    last = fdStats.evalSlices - 1;
    fdStats.usedSlices = fdStats.evalSlices - 2;
    quot = quot - 1;
end

if isempty(fdData)
  fdStats.globalFD = NaN;
else
  fdStats.globalFD = nanmean(fdData);
end

basalSlices = [first:first+quot-1];
apicalSlices = [first+quot+rem:last];

if isempty(fdData(apicalSlices))
  fdStats.meanApicalFD = NaN;
  fdStats.maxApicalFD  = NaN;
else
  fdStats.meanApicalFD = nanmean(fdData(apicalSlices));
  fdStats.maxApicalFD  = nanmax(fdData(apicalSlices));  
end    

if isempty(fdData(basalSlices))
  fdStats.meanBasalFD = NaN;
  fdStats.maxBasalFD  = NaN;
else
  fdStats.meanBasalFD = nanmean(fdData(basalSlices));
  fdStats.maxBasalFD  = nanmax(fdData(basalSlices));
end

File: python/src/fdStatistics_old.py
import numpy as np

def fd_statistics_old(fd_data, discard_mode):
    eval_slices = len(fd_data)
    quot, rem = divmod(eval_slices, 2)

    if not discard_mode:
        first = 1
        last = eval_slices
        used_slices = eval_slices
    else:
        first = 2
        last = eval_slices - 1
        used_slices = eval_slices - 2
        quot -= 1

    global_fd = np.nanmean(fd_data) if fd_data else np.nan

    basal_slices = range(first, first + quot)
    apical_slices = range(first + quot + rem, last + 1)

    mean_apical_fd = np.nanmean(fd_data[apical_slices]) if fd_data[apical_slices] else np.nan
    max_apical_fd = np.nanmax(fd_data[apical_slices]) if fd_data[apical_slices] else np.nan

    mean_basal_fd = np.nanmean(fd_data[basal_slices]) if fd_data[basal_slices] else np.nan
    max_basal_fd = np.nanmax(fd_data[basal_slices]) if fd_data[basal_slices] else np.nan

    return {
        "eval_slices": eval_slices,
        "used_slices": used_slices,
        "global_fd": global_fd,
        "mean_apical_fd": mean_apical_fd,
        "max_apical_fd": max_apical_fd,
        "mean_basal_fd": mean_basal_fd,
        "max_basal_fd": max_basal_fd,
    }
